Feature/create user settings page #281

Vansnoden opened the pull request.

ray-millward-tessella commented:
Hi Salvador, there are a lot of deleted files in the pull request that are needed by the tile server - you need to revert these changes. Similarly there are changes to files outside the UI - these shouldn't have changed and so should be reverted too.

The UI files you've changed should have corresponding unit tests and the build is currently failing because you need to run linting and fix the errors (most can be done automatically with `npm run lint:fix`)

andykitch commented:
Hi Salvador, thanks for addressing the above comments. Just had a look at the page locally - it looks good, nice work. A few comments on the code:

- Please can you remove the commented-out code?
- The files in the pages folder should just be wrappers around components, which live in the components folder. See `upload.tsx` for an example
- The new files will need unit tests - probably not the pages file, as it'll just be a wrapper around a component, but definitely the component file. There are lots of examples in the codebase.

Let me know if you want a hand with any of that.

Vansnoden commented:
> Hi Salvador, thanks for addressing the above comments. Just had a look at the page locally - it looks good, nice work. A few comments on the code:
> 
> * Please can you remove the commented-out code?
> * The files in the pages folder should just be wrappers around components, which live in the components folder. See `upload.tsx` for an example
> * The new files will need unit tests - probably not the pages file, as it'll just be a wrapper around a component, but definitely the component file. There are lots of examples in the codebase.
> 
> Let me know if you want a hand with any of that.

Well received

andykitch commented on src/UI/pages/user_settings.tsx:
```diff
@@ -30,94 +31,12 @@ function UserSettings() {
         </main>
       </div>
     );
   } else {
```
Do we need this if statement? If a user isn't logged in, they should be redirected to the login page as in line26, yes?

andykitch commented:
We'll need some unit tests for the `UserSettingsForm` component (you might be working on these anyway, just thought I'd check)

andykitch commented:
Test looks good, especially if you can get the router test in there too. Can you rename the test file to `UserSettingForm.test.tsx` please, to be consistent with the rest of the code?
